### PR TITLE
[sw] Promulgate better use of standard headers.

### PR DESCRIPTION
**Pull request by mcy**

### CODE REVIEW

File: sw/device/lib/common.h [hunk 1]
```diff
@@ -45,8 +38,8 @@ static const unsigned long UART_BAUD_RATE = 230400;
 #define BITLENGTH(X) \
   ((BITLENGTH_5(BITLENGTH_4(BITLENGTH_3(BITLENGTH_2(BITLENGTH_1(X)))))) & 0x7f)
 
-void *memcpy(void *restrict dest, const void *restrict src, size_t n);
-void *memset(void *dest, int val, size_t n);
```
Comment (gkelly) [inline sw/device/lib/common.h]: Isn't this spec'd as size_t?

Comment (mcy) [reply sw/device/lib/common.h]: I was being lazy here; I just wnated to get rid of the headers in common.h.

Comment (gkelly) [reply sw/device/lib/common.h]: Checking the spec this looks like it should actually be `rsize_t`, which I think is missing from the freestanding headers you implemented.  I don't think we should deviate here from the spec.

Comment (mcy) [reply sw/device/lib/common.h]: Uh, where are you reading this? There's no such thing as `rsize_t` in C11.

Comment (gkelly) [reply sw/device/lib/common.h]: K.3.3 defines `rsize_t`, but you're right `memcpy_s` is defined as taking `rsize_t` not `memcpy`.  I think you should pull in `stdint.h` for the definition here though.

Comment (mcy) [reply sw/device/lib/common.h]: Right, except that common.h no longer defines `memcpy` anymore, because my other PR landed. =P

Comment (gkelly) [reply sw/device/lib/common.h]: Well that solves that. 😋

**Pull request merged**